marshmallow_sqlalchemy/fields.py

```diff
@@ -87,26 +92,48 @@ def _serialize(self, value, attr, obj):
         return ret if len(ret) > 1 else list(ret.values())[0]
 
     def _deserialize(self, value, *args, **kwargs):
+        """Deserialize a serialized value to a model instance.
+
+        If the parent schema is transient, create a new (transient) instance.
+        Otherwise, attempt to find an existing instance in the database.
+        :param value: The value to deserialize.
+        """
         if not isinstance(value, dict):
             if len(self.related_keys) != 1:
                 self.fail('invalid', value=value, keys=[prop.key for prop in self.related_keys])
             value = {self.related_keys[0].key: value}
-        query = self.session.query(self.related_model)
+        if self.transient:
+            return self.related_model(**value)
+        if not self.session:
```

Comment:
I think this conditional is unnecessary, since this is checked in `Schema.load`.